platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java

```diff
@@ -289,26 +264,25 @@ public int delete(String type, String cql) throws PersistenceException {
         return idsToDelete.size();
     }
 
-    private SolrClient getSolrCore(String storeName) {
-        if (coreSolrClients.containsKey(storeName)) {
-            LOGGER.debug("Returning core {} from map of coreSolrClients", storeName);
-            return coreSolrClients.get(storeName);
+    private SolrClient getSolrClient(String storeName) {
```

Comment:
If we change the code to use a `ConcurrentHashMap<String, Future<SolrClient>>` and a separate `ConcurrentHashMap<SolrClient>`, then the potential concurrency issues might disappear. Let's see what the code looks like after that change is in.